status-im/status-react
# Push notifications: Spike for Android (and partial iOS)

Progress for push notifications spike for Android. Related to https://github.com/status-im/status-react/issues/1599 and achieves (1) "ability to manually trigger a push notification." by exposing an FCMToken.

PR only for devs to tests, nothing user-facing here. Mostly sanity check, but welcome to check actual notifications too.

### Summary:
- Print FCMToken which can be used to send PNs from Firebase Console
- Include react-native-fcm and related packages
- Update capabilities for iOS
- google service config files for Android and iOS

### Steps to test:

iOS no crash test:
- Open Status on iOS, make sure it doesn't crash. If using device, note whether it asks for permissions or not.

Android emulator test:
- Open Status on Android, if using Android ensure you have google-play-services installed. Using Genymotion this can be achieved with 2.10 and Open Gapps installed (https://github.com/opengapps/opengapps) - if you have issues PM me.
- Run `adb logcat | grep token` and note token
- Put app in background and go to Firebase Console (https://firebase.google.com/docs/cloud-messaging/) to trigger notification to that specific device.

status: ready
